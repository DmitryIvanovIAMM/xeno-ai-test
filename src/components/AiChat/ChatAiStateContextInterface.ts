'use client';

import { createContext, Dispatch, SetStateAction } from 'react';
import { ChatAiState, defaultChatAiState } from '@/Interfaces/interfaces';

export interface ChatAiStateContextInterface {
  chatAiState: ChatAiState;
  setChatAiState: Dispatch<SetStateAction<ChatAiState>>;
}

const defaultChatAiStateContext: ChatAiStateContextInterface = {
  chatAiState: defaultChatAiState,
  setChatAiState: () => {},
};

export const ChatAiStateContext =
  createContext<ChatAiStateContextInterface>(defaultChatAiStateContext);
