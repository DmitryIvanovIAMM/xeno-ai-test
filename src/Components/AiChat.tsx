'use client';

import ChatInput from '@/Components/ChatInput';
import ChatHistory from '@/Components/ChatHistory';
import React from 'react';
import { defaultChatAiState } from '@/Interfaces/interfaces';
import { ChatAiStateContext } from '@/Components/ChatAiStateContextInterface';

export default function AiChat() {
  const [chatAiState, setChatAiState] = React.useState(defaultChatAiState);

  return (
    <ChatAiStateContext.Provider value={{ chatAiState, setChatAiState }}>
      <h1>AI Chat</h1>
      <ChatHistory />
      <ChatInput />
    </ChatAiStateContext.Provider>
  );
}
