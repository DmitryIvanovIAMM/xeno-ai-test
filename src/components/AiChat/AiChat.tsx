'use client';

import ChatInput from '@/components/ChatInput/ChatInput';
import ChatHistory from '@/components/ChatHistory/ChatHistory';
import React from 'react';
import { defaultChatAiState } from '@/Interfaces/interfaces';
import { ChatAiStateContext } from '@/components/AiChat/ChatAiStateContextInterface';

export default function AiChat() {
  const [chatAiState, setChatAiState] = React.useState(defaultChatAiState);

  return (
    <ChatAiStateContext.Provider value={{ chatAiState, setChatAiState }}>
      <ChatHistory />
      <ChatInput />
    </ChatAiStateContext.Provider>
  );
}
