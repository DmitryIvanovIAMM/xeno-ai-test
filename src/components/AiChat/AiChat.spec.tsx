import { render } from '@testing-library/react';
import AiChat from '@/components/AiChat/AiChat';

const setup = async () => {
  const container = render(<AiChat />);
  return {
    container,
  };
};

describe('AiChat component', () => {
  it('should render correctly', async () => {
    const { container } = await setup();

    expect(container).toMatchSnapshot();
  });

  it('should contains structural information', async () => {
    const { container } = await setup();

    expect(container.container).toHaveTextContent('Allow mock responses');
    container.getByRole('button', {
      name: 'Send query',
    });
  });
});
