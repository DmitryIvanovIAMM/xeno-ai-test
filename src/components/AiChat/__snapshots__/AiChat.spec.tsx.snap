// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AiChat component should render correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container"
      >
        <div />
      </div>
      <div
        class="container"
      >
        <div>
          <label
            class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-19opgx6-MuiFormControlLabel-root"
          >
            <span
              class="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium css-imrjgg-MuiButtonBase-root-MuiCheckbox-root"
            >
              <input
                class="PrivateSwitchBase-input css-j8yymo"
                data-indeterminate="false"
                type="checkbox"
              />
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
                data-testid="CheckBoxOutlineBlankIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                />
              </svg>
            </span>
            <span
              class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-rizt0-MuiTypography-root"
            >
              Allow mock responses
            </span>
          </label>
        </div>
      </div>
      <div
        class="container"
      >
        <div
          class="MuiFormControl-root MuiFormControl-marginDense MuiFormControl-fullWidth MuiTextField-root css-xanwcj-MuiFormControl-root-MuiTextField-root"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-19qnlrw-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="false"
            for="outlined-multiline-static"
            id="outlined-multiline-static-label"
          >
            Put question here: 
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-1blp12k-MuiInputBase-root-MuiOutlinedInput-root"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiOutlinedInput-input css-16wblaj-MuiInputBase-input-MuiOutlinedInput-input"
              id="outlined-multiline-static"
              type="text"
              value=""
            />
            <fieldset
              aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1ll44ll-MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="css-81qg8w"
              >
                <span>
                  Put question here: 
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <div
          class="button"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary Mui-disabled MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary css-1abvrni-MuiButtonBase-root-MuiButton-root"
            disabled=""
            style="height: 58px;"
            tabindex="-1"
            type="button"
          >
            Send query
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container"
    >
      <div />
    </div>
    <div
      class="container"
    >
      <div>
        <label
          class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-19opgx6-MuiFormControlLabel-root"
        >
          <span
            class="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium css-imrjgg-MuiButtonBase-root-MuiCheckbox-root"
          >
            <input
              class="PrivateSwitchBase-input css-j8yymo"
              data-indeterminate="false"
              type="checkbox"
            />
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
              data-testid="CheckBoxOutlineBlankIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
              />
            </svg>
          </span>
          <span
            class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-rizt0-MuiTypography-root"
          >
            Allow mock responses
          </span>
        </label>
      </div>
    </div>
    <div
      class="container"
    >
      <div
        class="MuiFormControl-root MuiFormControl-marginDense MuiFormControl-fullWidth MuiTextField-root css-xanwcj-MuiFormControl-root-MuiTextField-root"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-19qnlrw-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink="false"
          for="outlined-multiline-static"
          id="outlined-multiline-static-label"
        >
          Put question here: 
        </label>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-1blp12k-MuiInputBase-root-MuiOutlinedInput-root"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiOutlinedInput-input css-16wblaj-MuiInputBase-input-MuiOutlinedInput-input"
            id="outlined-multiline-static"
            type="text"
            value=""
          />
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-1ll44ll-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-81qg8w"
            >
              <span>
                Put question here: 
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <div
        class="button"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary Mui-disabled MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary css-1abvrni-MuiButtonBase-root-MuiButton-root"
          disabled=""
          style="height: 58px;"
          tabindex="-1"
          type="button"
        >
          Send query
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
